src/policy/protect_payload.rs

```diff
@@ -99,10 +105,25 @@ impl PolicyModule for ProtectPayloadPolicy {
 
         // Step 2: Save sensitives privileged registers
         self.confidential_values_payload = get_confidential_values(ctx);
-        self.write_confidential_registers(ctx, false)
+        self.write_confidential_registers(ctx, false);
+
+        // Step 3: Lock memory
+        mctx.pmp.set(
+            mctx.devices.len() + 2,
+            build_napot(0x80400000, 0x80000).unwrap(),
+            pmpcfg::NAPOT,
+        );
+
+        unsafe {
+            Arch::write_pmp(&mctx.pmp).flush();
+        }
```

Comment:
And with the new PMP refactor we would not need to flush, right?

Reply:
Same comment, but indeed there is no need.